Allow mail_handler to find case insensitive keys

oliverguenther opened the pull request:
https://community.openproject.com/wp/32251

oliverguenther committed:
Allow mail_handler to find case insensitive keys

ulferts commented:
While I agree that this feature is desirable I see problems because of our current uniqueness checks. 
There could be two Types/Versions/... with almost identical names except for differences in upcase/downcase. Referencing those becomes non deterministic now. While I do not want to say that this is a common use case, we should prevent such clashes. 

How about we allow the behaviour but check whether more than one entry matches the lowercased key. In those cases, we bail, but accept the change as long as it is deterministic.

oliverguenther commented:
Statuses are actually already tested for lowercase. How about we simply add a validation to avoid case-differency-only types and versions as well? I'm fine with your approach for the course of this PR however

ulferts commented:
Turning the validations case insensitive is fine with me @oliverguenther.

oliverguenther committed:
Change case insensitive validations to false

oliverguenther commented:
I changed the validations to case-insensitive and created a Feature so it turns up in the changelog
https://community.openproject.com/wp/32513

I also changed the branch to dev to avoid merging this in a patch release

ulferts commented on app/models/enumeration.rb:
```diff
@@ -40,9 +40,9 @@ class Enumeration < ActiveRecord::Base
 
   validates_presence_of :name
-  validates_uniqueness_of :name, scope: %i(type project_id)
-  validates_length_of :name, maximum: 30
```
If this is an intended change, we need to also change the type in the database.

oliverguenther replied:
No this was intended, thanks for catching

ulferts reviewed (changes requested):
Admittedly being paranoid here, but we probably need a migration for this change to alter formerly valid enumerations/categories/... that now became duplicates.

oliverguenther committed:
Remove duplicate type_standard "None" and "none" in legacy spec

oliverguenther committed:
Avoid overridden submit_options in context

ulferts reviewed (approved):
Looks good :+1:

We talked about the necessity of adding migrations and agreed to not need them.
